daemons/controld/controld_join_dc.c

```diff
@@ -539,9 +539,10 @@ do_dc_join_filter_offer(long long action,
 
         if (pcmk__get_schema(validation) == NULL) {
             crm_err("Rejecting join-%d request from %s (with first CIB "
-                    "generation) due to unknown schema version %s "
-                    QB_XS " ref=%s",
-                    join_id, join_from, pcmk__s(validation, "(missing)"), ref);
+                    "generation) due to %s schema version %s " QB_XS " ref=%s",
+                    join_id, join_from,
+                    ((validation == NULL)? "missing" : "unknown"),
+                    pcmk__s(validation, ""), ref);
```

Comment:
leaves an extra space character in the message if `validation` is `NULL` , but it only really shows up in the verbose form; not worth changing

Reply:
right, and it should never really happen anyway